### PR TITLE
Document anonymous access and limitations

### PR DESCRIPTION
**Pull request by deviantintegral**
# Proposed Changes

This follows up from my question at #100 with notes on how to set up local access and limitations for cloud access. [Link to the rendered version in this PR](https://github.com/hassio-addons/addon-grafana/blob/9af451655e42a13e8a8a43922eef3c4f62e866b7/grafana/DOCS.md#anonymous-access).

### CODE REVIEW

File: grafana/DOCS.md [hunk 1]
```diff
@@ -113,6 +113,28 @@ for Home Assistant.
 
 1. Hit Save & Test
 
+## Anonymous Access
+
+Anonymous access on a local network is possible by exposing the port in the
+addon config, along with setting the following environment variables:
```
Comment (frenck) [inline grafana/DOCS.md:119]:
```suggestion
add-on config, along with setting the following environment variables:
```

File: grafana/DOCS.md [hunk 2]
```diff
@@ -113,6 +113,28 @@ for Home Assistant.
...
+    value: "true"
+  - name: GF_AUTH_ANONYMOUS_ACCESS_ORG_NAME
+    value: "Main Org."
+  - name: GF_AUTH_ANONYMOUS_ACCESS_ORG_ROLE
+    value: "Viewer"
+```
+
+It is not possible to enable anonymous or non-administrator access with Home
+Assistant Cloud. This includes embedding Grafana resources with an iframe or
+rendered image inside of a dashboard. For more details see
```
Comment (frenck) [inline grafana/DOCS.md:133]:
```suggestion
rendered image inside of a dashboard. For more details see:
```

File: grafana/DOCS.md [hunk 3]
```diff
@@ -113,6 +113,28 @@ for Home Assistant.
...
+["“401: Unauthorized” iframe CARD of Grafana not working"](https://community.home-assistant.io/t/401-unauthorized-iframe-card-of-grafana-not-working/148017)
+and
+[Anonymous login not working, Grafan add-on 3.0.0 #55](https://github.com/hassio-addons/addon-grafana/issues/55).
```
Comment (frenck) [inline grafana/DOCS.md:136]:
```suggestion

- ["“401: Unauthorized” iframe CARD of Grafana not working"](https://community.home-assistant.io/t/401-unauthorized-iframe-card-of-grafana-not-working/148017)
and
- [Anonymous login not working, Grafana add-on 3.0.0 #55](https://github.com/hassio-addons/addon-grafana/issues/55).
```

**Review by frenck — CHANGES_REQUESTED** [review]

**Comment by frenck** [comment]
Additionally requested a review from @sinclairpaul, he knows the most about this.

**Comment by sinclairpaul** [comment]
Grafana supports publishing snapshots I believe, which automatically sets up anonymous access, wouldn't that be the preferred approach rather than essentially opening the instance to anyone?

I haven't looked at it in detail, so may need some config adjustment to function, but the details can be found at:

https://grafana.com/docs/grafana/latest/sharing/share-panel/

I will also add, I'm really not a fan of iFraming anything, and as browser security becomes stricter it becomes more challenging.  We should probably at least add a note around samesite config .

**Comment by deviantintegral** [comment]
Thanks for the review.

> Grafana supports publishing snapshots I believe, which automatically sets up anonymous access, wouldn't that be the preferred approach rather than essentially opening the instance to anyone?

The issue with that feature is that it limits the data to what the snapshot was created with. It doesn't expose data logged after the snapshot was created. The other option is embeds, but those are tied to your authentication state.

The share panel offers embeds which are meant to be accessed through an iframe, but they're tied to the authentication state.

**Comment by deviantintegral** [comment]
Re iframes, how about I link that to https://grafana.com/docs/grafana/latest/sharing/share-panel/#embed-panel so it's clear that's what I'm referring to? I assume for same-site that Grafana explicitly doesn't set that given their examples use iframes.

**Comment by sinclairpaul** [comment]
Thanks, if the snapshot links aren't updated it makes sense.

My only last comment is linking to forum posts that could vanish maybe??, other than that I am good.

**Comment by frenck** [comment]
Yeah, I agree with @sinclairpaul there.

**Comment by sinclairpaul** [comment]
Just one last thought, Grafana does offer API auth, does this work for dashboards?

https://grafana.com/docs/grafana/latest/http_api/auth/

Sorry it took me a while to find it (I thought it was there somewhere), it looks like it can be passed as Basic Auth, and may be preferable than opening the whole instance up.

**Comment by deviantintegral** [comment]
If the point is to give access to viewers to the Grafana dashboards, I don't think HTTP basic will help. For example, Safari no longer passes through credentials in a URL as HTTP basic headers, requiring the user to put them in again.

**Comment by deviantintegral** [comment]
I think we're pretty close here! Let me know if there's anything else to do.

File: grafana/DOCS.md [hunk 4]
```diff
@@ -113,6 +113,26 @@ for Home Assistant.
 
 1. Hit Save & Test
 
+## Anonymous Access
+
+Anonymous access on a local network is possible by exposing the port in the
+add-on config, along with setting the following environment variables:
```
Comment (frenck) [inline grafana/DOCS.md:119]:
```suggestion
Anonymous access on a local network is possible by exposing the port in the
add-on configuration, along with setting the following environment variables:
```

**Pull request merged**